sphinx_gallery/gen_rst.py

```diff
@@ -1641,6 +1643,8 @@ def _get_class(gallery_conf, key):
                 f"{what} must be a fully qualified name string, could not import "
                 f"{attr} from {mod}"
             )
+        if not is_builtin_alias:
+            val = FunctionSortKey(val)
```

Comment:
just to spell out the logic of this function:

- if it's a string alias of a builtin SortKey class, convert to FQN string
- then, if it's a string, check for ".", split, and try to import
- if that worked, wrap the imported func in `FunctionSortKey` (unless it's already a built-in SortKey class)